qiskit_dynamics/solvers/solver_functions.py

```diff
@@ -273,6 +282,17 @@ def rhs(t, y):
 
     if method == "scipy_expm":
         results = scipy_expm_solver(solver_generator, t_span, y0, t_eval=t_eval, **kwargs)
+    if method == "lanczos_diag":
+        if "sparse" not in generator.evaluation_mode:
+            warn(
+                "lanczos_diag must be used with a generator in sparse mode for better performance.",
+                category=Warning,
+                stacklevel=5,
+            )
+            # raise QiskitError("lanczos_diag must be used with a generator in sparse mode.")
+        if not is_hermitian(1j * solver_generator(1.12)):
```

Comment:
We discussed this before but right now I'm leaning towards not checking this condition since it can't be checked for all time anyway. 

I think the best that can be done is: 
- to check if its a `LindbladModel` or a `GeneratorModel`, and raise an error in those cases, something like "Lanczos solver can only be used for HamiltonianModel or function-based anti-Hermitian generators.".
- Be very clear in the documentation of the solver that it only works for anti-Hermitian generators.